Brokerages/GDAX/GDAXBrokerage.Messaging.cs

```diff
@@ -61,9 +60,10 @@ internal enum GdaxEndpointType { Public, Private }
         private int _pendingLeanMarketOrderId;
 
         /// <summary>
-        /// Rest client used to call missing conversion rates
+        /// QC API client used to fetch missing conversion rates
         /// </summary>
-        public IRestClient RateClient { get; set; }
+        public IApi RateClient { get; set; } = new Api.Api();
```

Comment:
Instead of requiring the entire API object (and also instead of 'newing' it up in this file, let's define an abstraction, something like `IConversionRateProvider` and accept it as a dependency in the constructor.

This will also remove the need to accept the qc user id and the qc user token in the constructor :)

There was a mock of this interface done on the multi-currency conversion PR #2060 

That interface was more focused on currencyA->currencyB conversions, no reason it can't be modified to accept currencies A and B in the `GetConversionRate()` method.